doc/specs/#607 - Commandline Arguments for the Windows Terminal.md

```diff
@@ -0,0 +1,724 @@
+in the codebase to enable this functionality in a way that users are expecting.
+The following is a suggestion of the individual changelists that could be made
+to iteratively work towards fulling implementing this funcionality.
+
+* [x] Refactor `ShortcutAction` dispatching into its own class
+  - Right now, the `AppKeyBindings` is responsible for triggering all
+    `ActionAndArgs` events, but only based upon keystrokes while the Terminal is
+    running. As we'll be re-using `ActionAndArgs` for handling startup events,
+    we'll need a more generic way of dispatching those events.
+* [x] Add a `SplitPane` `ShortcutAction`, with a single parameter `split`,
+  which accepts either `vertical` or `horizontal`
+  - Make sure to convert the legacy `SplitVertical` and `SplitHorizontal` to use
+    `SplitPane` with that arg set appropriately.
+* [x] Add a `TerminalParameters` winrt object to `NewTabArgs` and `SplitPane`
+  args. `TerminalParameters` will include the following properties:
+
+```c#
+runtimeclass TerminalParameters {
+    String ProfileName;
+    String ProfileGuid;
+    String StartingDirectory;
+    String Commandline;
+}
+```
+  - These represent the arguments in `[terminal_parameters]`. When set, they'll
+    both `newTab` and `splitPane` will accept [`profile`, `guid`, `commandline`,
+    `startingDirectory`] as optional parameters, and when they're set, they'll
+    override the default values used when creating a new terminal instance.
+    - `profile` and `guid` will be used to look up the profile to create by
+      `name`, `guid`, respectively, as opposed to the default profile.
+    - The others will override their respective properties from the
+      `TerminalSettings` created for that profile.
+* [ ] Add an optional `"percent"` argument to `SplitPane`, that enables a pane
+  to be split with a specified percent of the parent pane.
+* [ ] Add support to `TerminalApp` for parsing commandline arguments, and
+  constructing a list of `ActionAndArgs` based on those commands
+  - This will include adding tests that validate a particular commandline
+    generates the given sequence of `ActionAndArgs`.
+  - This will _not_ include _performing_ those actions, or passing the
+    commandline from the `WindowsTerminal` executable to the `TerminalApp`
+    library for parsing. This change does not add any user-facing functional
+    behavior, but is self-contained enough that it can be its own changelist,
+    without depending upon other functionality.
+* [ ] When parsing a `new-tab` command, configure the `TerminalApp::AppLogic` to
+  set some initial state about itself, to handle the `new-tab` arguments
+  [`--initialPosition`, `--maximized`, `--initialRows`, `--initialCols`]. Only
+  set this state for the first `new-tab` parsed. These settings will overwrite
+  the corresponding global properties on launch.
+  - at this time, this state is still not settable externally by the user.

```

Comment:
May not mention this, as it dates the spec and it may not be removed when it's updated.